src/language-html/embed.js

```diff
@@ -3,53 +3,89 @@
 const { hasNewlineInRange } = require("../common/util");
 const {
   builders: { hardline, concat },
-  utils: { stripTrailingHardline, removeLines }
+  utils: { removeLines }
 } = require("../doc");
 
 function embed(path, print, textToDoc, options) {
   const node = path.getValue();
 
   switch (node.type) {
     case "text": {
-      const parent = path.getParentNode();
-      // Inline JavaScript
-      if (
-        parent.type === "script" &&
-        ((!parent.attribs.lang && !parent.attribs.lang) ||
-          parent.attribs.type === "text/javascript" ||
-          parent.attribs.type === "application/javascript")
-      ) {
-        const parser = options.parser === "flow" ? "flow" : "babylon";
-        const doc = textToDoc(getText(options, node), { parser });
-        return concat([hardline, doc]);
+      const node = path.getValue();
+      const parentNode = path.getParentNode();
+
+      if (!parentNode) {
+        return null;
+      }
+
+      let parser;
+
+      if (parentNode.type === "style") {
+        if (Object.keys(parentNode.attribs).length === 0) {
+          parser = "css";
+        }
+
+        const typeAttr = parentNode.attribs["type"];
+
+        if (typeAttr === "text/css") {
+          parser = "css";
+        } else if (typeAttr === "text/x-scss") {
+          parser = "scss";
+        } else if (typeAttr === "text/less") {
+          parser = "less";
+        } else {
+          const langAttr = parentNode.attribs["lang"];
+
+          if (langAttr === "postcss") {
+            parser = "css";
+          } else if (langAttr === "scss") {
+            parser = "scss";
+          } else if (langAttr === "less") {
+            parser = "less";
+          } else {
+            parser = "css";
+          }
+        }
       }
 
-      // Inline TypeScript
-      if (
-        parent.type === "script" &&
-        (parent.attribs.type === "application/x-typescript" ||
-          parent.attribs.lang === "ts")
-      ) {
-        const doc = textToDoc(
-          getText(options, node),
-          { parser: "typescript" },
-          options
-        );
-        return concat([hardline, doc]);
+      if (parentNode.type === "script") {
+        if (Object.keys(parentNode.attribs).length === 0) {
+          parser = "babylon";
+        }
+
+        const typeAttr = parentNode.attribs["type"];
+
+        if (
```

Comment:
Don't forget to add `parser:json` e.g for common `<script type="application/ld+json">` tags